Orange/data/table/base.py

```diff
@@ -0,0 +1,1409 @@
+        writer = FileFormat.writers.get(ext)
+        if not writer:
+            desc = FileFormat.names.get(ext)
+            if desc:
+                raise IOError("Writing of {}s is not supported".format(desc.lower()))
+            else:
+                raise IOError("Unknown file name extension.")
+        writer.write_file(filename, self)
+
+    def __getitem__(self, item):
+        # if selecting a column subset, we need to transfer weights so they don't just disappear
+        # only do this for multiple column selection, which returns a DataFrame by contract
+        if isinstance(item, (Sequence, pd.Index, np.ndarray)) and not isinstance(item, str) \
+                and all(isinstance(i, str) for i in item) \
+                and any(ic not in item for ic in self._INTERNAL_COLUMN_NAMES):
+            item = list(item) + [ic for ic in self._INTERNAL_COLUMN_NAMES if ic not in item]
+        return super().__getitem__(item)
+
+    def __setitem__(self, key, value):
+        # if the table has an empty index and we're inserting a the first row,
+        # the index would be created by pandas automatically.
+        # we want to maintain unique indices, so we override the index manually.
+        # we also need to set default weights, lest they be NA
+        new_index_and_weights = len(self.index) == 0
+
+        # SOMEWHAT PANDAS CONTRACT BREAKAGE:
+        # the pandas default behaviour when adding a new column as a Series is that
+        # the indexes are inner-joined: only the elements at the indices that exist
+        # in the current table are actually set, other elements are NA
+        # for easier handling of Orange behaviour, we effectively ignore the index
+        # on the series if the series is of the same length (like when adding a new column)
+        # to just merge the column into the table
+        if isinstance(value, pd.Series) and len(value) == len(self) and not new_index_and_weights:
+            value.index = self.index
+        super().__setitem__(key, value)
+
+        if new_index_and_weights:
+            new_id = self._new_id(len(self), force_list=True)
+            self.index = new_id
+            # super call because we'd otherwise recurse back into this
+            super().__setitem__(self._WEIGHTS_COLUMN, 1)
+
+    def __iter__(self):
+        """Iterate over the rows of this TableBase as SeriesBase, breaking the pandas contract.
+
+        Returns
+        -------
+        generator
+            A generator of rows as SeriesBase objects.
+
+        Examples
+        --------
+            >>> iris = Table('iris')
+            >>> for row in iris.iloc[:5]:
+            >>>     print(row)
+
+        Notes
+        -----
+        This breaks the pandas contract! Pandas iterates over column names by default.
+        However, this only breaks __str__ and __repr__, which are reimplemented anyway.
+        """
+        for _, row in self.iterrows():
+            yield row
+
+    def __str__(self):
```

Comment:
So I'm kind of (100%) against this. I'd like to see an example of representation that is more Orange or user friendly than the current pandas' representation.

Pandas provides some thirty tunables in `pd.options.display` that print the stuff as beautifully as the user prefers. I'd hate to account for/maintain any attempts to mimic that. Especially when the implementation is poor and the result looks poorer. Perhaps you can achieve friendly-enough representation by tweaking some of those tunables.

A nice alternative is perhaps to construct a [multi-index](http://pandas.pydata.org/pandas-docs/stable/advanced.html) for column roles:
```py
roles = [(role, var.name)
         for role, vars in (('feature', self.domain.attributes),
                            ('class', self.domain.class_vars),
                            ('meta-data', self.domain.metas))
         for var in vars]
columns = pd.MultiIndex.from_tuples(roles, names=['feature role', 'feature name'])
return str(self._constructor(self, columns=columns))
```